spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -314,7 +315,29 @@ case class CreateDeltaTableCommand(
             txn.snapshot.domainMetadata, newDomainMetadata)
         taggedCommitData = taggedCommitData.copy(actions = newActions)
       }
-      val op = getOperation(txn.metadata, isManagedTable, Some(options),
+
+      def schemaContainsExistsColumnMetadata(schema: StructType): Boolean = {
+        schema.exists(
+          _.metadata.contains(ResolveDefaultColumnsUtils.EXISTS_DEFAULT_COLUMN_METADATA_KEY))
+      }
+      val metadataForOp =
```

Comment:
This logic could use a comment to explain _why_ we're doing it this way. IIUC we need to preserve the metadata on the source because the defaults are applied on read retroactively in the "exists default" case? So we keep the metadata on the source and we remove it from the target table, which makes it ineligible for `DeltaTableUtils.removeInternalWriterMetadata` and `ColumnWithDefaultExprUtils.removeDefaultExpressions`.